wiki/Ranking_Criteria/osu!taiko/en.md

```diff
@@ -3,46 +3,161 @@ osu!taiko
+- **Spinner:** A round element that requires the player to alternate between hitting Don/Kat for a determined number of times, which is shown in the center of the spinner. This number depends on the Overall Difficulty of the map as well as the length of the spinner.
+- **Bar Line:** A line that appears on the playfield and determines the start of a measure if the BPM is higher than 75. If the BPM is smaller or equal to 75 bar lines will be drawn on every beat.
+- **Overlap:** A note that partially or totally covers other note(s) in the playfield.
+- **Rest Moment:** A period of time without notes used specifically to allow the player to rest their hands and prepare for the upcoming patterns.
+- **Stream:** Consecutive circles grouped together. Usually snapped to ¼ beats.
+- **Snapping:** Timeline tick where an object is placed.
+- **Variable Snapping:** A combination of multiple different ways to snap notes within a short span of time due to the song’s fluctuating nature at that point.
+- **Slider Velocity:** Speed at which the notes/drumrolls/spinners/bar lines move horizontally from right to left through the playfield. Base slider velocity can be controlled in the timing panel and additional changes can be made through inherited (green) timing points.
+- **Smooth Slider Velocity Changes:** A mechanic that enables gradual transition from lower to higher / from higher to lower Slider Velocity sections. Transition notes with varying Slider Velocities are used to achieve this effect.
+- **Improvisation:** Having more notes than the actual song is providing.
+- **Taiko Template Background:** Background images that simulate authentic Taiko no Tatsujin gameplay. They usually include a black bar showing artist and song title in white beneath the playfield.
 
-Rules
--------
+## General
 
-All rules are exactly that: **rules**. They are **not** guidelines and may **not** be broken under **any** circumstance.
+### Rules
 
--   **Finish notes must not be in the middle or at the beginning of a 1/4 or above snapped pattern.** This is to ensure playability of all patterns, as finisher notes in the middle or at the beginning of a 1/4 or above snapped pattern have been deemed as very issuesome to catch.
+All rules are exactly that: **rules**. They are **not** guidelines and must **not** be broken under **any** circumstance.
 
-Guidelines
---------------
+- **Each note must have its color clearly distinguishable from the previous and upcoming notes.**
+- **Every note must be clearly assignable to a musical layer or layer unit it tries to represent, be it to enhance a layer the song provides, or be it an additional layer improvised by the mapper.** Do not improvise in a fashion that shifts the pace, contradicts the general music movement or misinterprets the song's current intensity. Improvising must either enhance a current layer of the song or add a new one. Otherwise, relation to the song is given up and this contradicts the main purpose of a rhythm game.
```

Comment:
"or be it an additional layer improvised by the mapper" -> remove "be it" from this part to avoid the repetition

Reply:
helps reenforce that point, doesn't hurt to have